src/components/heatmap/heatmap.tsx

```diff
@@ -84,32 +91,48 @@ const Heatmap = ({
         'hide_null_choice': true
       }]
     });
-
-    ConfigService.ERMrest.resolve(ermrestURI.ermrestUri, header).then((response: any) => {
-      const reference = response.contextualize.compact;
-      verifyConfiguration(reference);
-      if (!configErrorsPresent) {
-        const sortBy = typeof Object(config).data.sortBy !== 'undefined' ? Object(config).data.sortBy : [];
-        const ref = reference.sort(sortBy);
-        setHeader((prevHeader: any) => ({ ...prevHeader, action: 'main' }));
-        const pcid = getQueryParams(location.href).pcid
-        const ppid = getQueryParams(location.href).ppid
-        if (pcid || ppid){
-          let tempHeader = {...header}
-          if(pcid)
-            tempHeader.pcid=pcid
-          if(ppid)
-            tempHeader.ppid=ppid
-          setHeader(() => (tempHeader));
+    const geneId = location.hash.split('/')[2];
+    // try to fetch the resolver link to see if the path resolves before sending the user
+    ConfigService.http.get(`/ermrest/catalog/2/entity/Common:Gene/${geneId}`,
+      { headers: header }).then((response: any) => {
+        setIsLoading((prevLoading) => !prevLoading);
+        if (response.data?.length > 0) {
+          if (response.data[0].Array_Data) {
+            ConfigService.ERMrest.resolve(ermrestURI.ermrestUri, header).then((response: any) => {
+              const reference = response.contextualize.compact;
+              verifyConfiguration(reference);
+              if (!configErrorsPresent) {
+                const sortBy = typeof Object(config).data.sortBy !== 'undefined' ? Object(config).data.sortBy : [];
+                const ref = reference.sort(sortBy);
+                setHeader((prevHeader: any) => ({ ...prevHeader, action: 'main' }));
+                const pcid = getQueryParams(location.href).pcid
+                const ppid = getQueryParams(location.href).ppid
+                if (pcid || ppid) {
+                  let tempHeader = { ...header }
+                  if (pcid)
+                    tempHeader.pcid = pcid
+                  if (ppid)
+                    tempHeader.ppid = ppid
+                  setHeader(() => (tempHeader));
+                }
+                ref.read(1000, header).then((page: any) => {
+                  readAll(page);
```

Comment:
Before calling `readAll`, we should do the check here to see if there is any "array data". `ref.read()` is making a get request to the `ermrestUri` path which is to the array data table. 

`Page` is an API that Aref and I designed so you can safely call `Page.length` to check how many rows are returned for the `ermrestUri` request.

```
if (page.length === 0) {
  throw new ChaiseError('Data not available', 'No heatmap data available for the given ID<br>');
}

readAll(page);
```

Make sure to test that this error is being thrown/set properly. You will have to make a change to the `catch()` block right after this to make sure this error is shown. I think the catch will ignore any thrown error and just blindly `setInvalidConfigs`.

Maybe something like:
```
}).catch((err: any) => {
  if (err) {
    throw err;
  } else {
    setInvalidConfigs(['Error while reading data from Ermrestjs']);
    setConfigErrorsPresent(true);
  }
});
```